[PR] 회원 검색 로직 서비스 레이어 위치 변경 및 api 통합

ehgur062300 opened the pull request:
- userService는 repository와의 연결 역할만으로 쓰임

ehgur062300 committed:
refactor: 회원 검색 로직 서비스 레이어 위치 변경

- userService는 repository와의 연결 역할만으로 쓰임

qkrtkdwns3410 commented on src/main/java/com/st/eighteen_be/user/service/UserService.java:
```diff
@@ -88,19 +88,10 @@ int findLikeCountById(Integer id) {
+
+    @Transactional
+    public List<UserInfo> findCreatedUsers(LocalDateTime searchDateTime) {
+        return userRepository.findAllByCreatedDateAfterOrLastModifiedDateAfter(searchDateTime, searchDateTime);
```
@Transactional   (readonly true) 부탁드립니당

어차피 외부에서 readonly false 걸면 읽기전용으로 동작하진 않을거같긴한데..

ehgur062300 replied:
실수였네요 수정했습니다!

qkrtkdwns3410 commented on src/main/java/com/st/eighteen_be/user/service/UserDtoService.java:
```diff
@@ -25,124 +24,143 @@
 @Slf4j
 @Service
 @RequiredArgsConstructor
-@Transactional
 public class UserDtoService {
```
트랜잭션 빠져도 되나요>?

ehgur062300 replied:
UserDtoService는 이미 트랜잭션이 걸려있는 userService를 호출하는 작업만 하기때문에 없어도 된다고 판단했습니다!
UserDtoService 전체에 트랜잭션을 걸어버리면 이중 트랜잭션이 발생할 거 같아서요!

qkrtkdwns3410 reviewed (changes requested):
확인했습니다

ehgur062300 committed:
refactor: 잘못된 트랜잭션 수정

ehgur062300 committed:
refactor: 불필요한 트랜잭션 제거

- 각 작업들 간의 데이터 일관성을 유지할 필요가 없고, 각각 트랜잭션 설정이 되어 있기 때문에 해당 메서드를 트랜잭션으로 묶을 필요 없다고 판단

ehgur062300 committed:
refactor: 잘못된 트랜잭션 설정 변경

- repository에 저장하는 로직이 있기 때문에 readOnly 설정을 false로 변경

ehgur062300 committed:
refactor: 회원 전체 조회 api와 카테고리 전체 조회 api 통합

- "전체" 카테고리 추가
